File: ImagePlatformFrontend/src/interfaces/Post.ts
import { Category } from "./Category";
import { User } from "./User";

export interface Post{
    imageUrl: string,
    datePublished: Date,
    views: number,
    downloads: number,
    description: string,
    id: string,
    isPremium: boolean,
    author: User,
    category: Category
}

File: ImagePlatformBackend/ImagePlatform/Models/Post.cs
using System.ComponentModel.DataAnnotations;

namespace ImagePlatform.Models
{
    public class Post
    {
        [Key]
        public Guid Id { get; set; }
        public string? ImageUrl{ get; set; }
        public DateTime DatePublished{ get; set; }
        public string Description { get; set; }
        public long Views { get; set; }
        public long Downloads { get; set; }
        public bool IsPremium { get; set; }
        public Category? Category { get; set; }
        public Guid? CategoryId { get; set; }
        public User Author { get; set; }
        public Guid AuthorId { get; set; }
    }
}
